File: apps/web/src/models/InventoryBox.ts
import { Product } from "./Product";

export interface InventoryBox {
  id: number;
  box_code: string;
  product_id: number;
  quantity: number;
  box_date: string;
  is_active: boolean;
  created_at: string;
  removed_at?: string | null;
  product: Product;
}

export interface InventoryBoxCreateDTO {
  product_id: number;
  quantity: number;
  box_date: string;
  box_code?: string | null;
}

export interface InventoryBoxUpdateDTO {
  product_id?: number;
  quantity?: number;
  box_date?: string;
  box_code?: string | null;
  is_active?: boolean;
}

File: apps/api/models/inventory_box.py
from datetime import datetime

from sqlalchemy import Boolean, Column, Date, DateTime, ForeignKey, Integer, String, true
from sqlalchemy.orm import relationship

from db.session import Base


class InventoryBox(Base):
    __tablename__ = "inventory_boxes"

    id = Column(Integer, primary_key=True, index=True)
    box_code = Column(String, nullable=False, unique=True, index=True)
    product_id = Column(Integer, ForeignKey("products.id"), nullable=False)
    quantity = Column(Integer, nullable=False)
    box_date = Column(Date, nullable=False, index=True)
    is_active = Column(Boolean, nullable=False, default=True, server_default=true())
    created_at = Column(DateTime, nullable=False, default=datetime.now)
    removed_at = Column(DateTime, nullable=True)

    product = relationship("Product")
    snapshot_items = relationship("InventorySnapshotItem", back_populates="box")
